go/arrow/array/numeric.gen.go.tmpl

```diff
@@ -82,6 +82,32 @@ func (a *{{.Name}}) setData(data *Data) {
 	}
 }
 
+func (a *{{.Name}}) ValueStr(i int) string {
+	if a.IsNull(i) {
+		return NullValueStr
+	}
+{{if or (eq .Name "Date32") (eq .Name "Date64") -}}
+	return a.values[i].ToTime().Format("2006-01-02")
+{{else if or (eq .Name "Time32") (eq .Name "Time64") -}}
+	return a.values[i].ToTime(a.DataType().(*{{.QualifiedType}}Type).Unit).Format("15:04:05.999999999")
```

Comment:
`Time32` and `Time64` also have `FormattedString` methods which take the unit. so this should be:

`return a.values[i].FormattedString(a.DataType().(*{{.QualifiedType}}Type).Unit)`